src/as_execution/mod.rs

```diff
@@ -15,82 +15,182 @@ use crate::settings::max_number_of_pages;
 use crate::tunable_memory::LimitingTunables;
 use crate::GasCosts;
 
-pub(crate) use common::*;
 pub(crate) use context::*;
 pub(crate) use error::*;
 
 #[derive(Clone)]
 pub enum RuntimeModule {
-    ASModule((ASModule, Engine)),
+    ASModule(ASModule),
 }
 
 impl RuntimeModule {
-    /// TODO: Dispatch module creation corresponding to the first bytecode byte
+    /// Dispatch module creation corresponding to the first bytecode byte
     ///
-    /// * (1) target AssemblyScript
+    /// * (1) TODO: target AssemblyScript (remove ident)
     /// * (2) TODO: target X
-    /// * (_) target AssemblyScript and use the full bytecode
-    pub fn new(bytecode: &[u8], limit: u64, gas_costs: GasCosts) -> Result<Self> {
+    /// * (_) target AssemblyScript
+    pub fn new(
+        bytecode: &[u8],
+        limit: u64,
+        gas_costs: GasCosts,
+        compiler: Compiler,
+    ) -> Result<Self> {
         match bytecode.first() {
-            Some(1) => Ok(Self::ASModule(ASModule::new(bytecode, limit, gas_costs)?)),
-            Some(_) => Ok(Self::ASModule(ASModule::new(bytecode, limit, gas_costs)?)),
+            Some(_) => Ok(Self::ASModule(ASModule::new(
+                bytecode, limit, gas_costs, compiler,
+            )?)),
             None => Err(anyhow!("Empty bytecode")),
         }
     }
+
+    /// Used compiler for the current module
+    pub fn compiler(&self) -> Compiler {
+        match self {
+            RuntimeModule::ASModule(module) => module.compiler.clone(),
+        }
+    }
+
+    /// Serialize a RuntimeModule
+    ///
+    /// TODO: set a module identifier for other types of sub modules.
```

Comment:
We don't want to have a byte right now setup to avoid having to create a new version of bytecode in the future ?

Reply:
It requires proper parsing to handle and we do not know what shape it will take. This would be premature implementation and I try to avoid this as much a I can. In any not having an ident now isn't a problem for the future since we could simply the default case to deser into this version of the module.

Reply:
Yes but what if the first byte of the bytecode is a `0` or `1` and so threat as a version in the future. ANyway, we can stay like that for now

Reply:
I think this is linked to this follow-up that I haven't tackled yet: https://github.com/massalabs/massa/issues/3598#issuecomment-1444097921
I have a branch with the changes for massa, but didn't put up a PR yet as I haven't made the changes to runtime. I think it's better for this PR to be oblivious to that change, and to make every change in the follow-up.